jax/experimental/host_callback.py

```diff
@@ -1959,43 +1958,41 @@ def barrier_wait(logging_name: Optional[str] = None):
       for this invocation. See `Debugging` in the module documentation.
   """
   logging_name = logging_name or ""
-  if logging.vlog_is_on(2):
-    logging.vlog(2, f"barrier_wait[{logging_name}]: start")
+  if logger.isEnabledFor(logging.DEBUG):
+    logger.debug("barrier_wait[%s]: start", logging_name)
 
   lock = threading.Lock()
   cv = threading.Condition(lock=lock)
   devices_at_barrier = []  # Protected by lock
   def barrier_tap_received(dev_idx, _):
     device = _callback_handler_data.devices[dev_idx]
-    if logging.vlog_is_on(2):
-      logging.vlog(
-          2,
-          f"barrier_wait[{logging_name}]: at barrier_tap for device {device} "
-          f". Thread {threading.current_thread()}")
+    if logger.isEnabledFor(logging.DEBUG):
+      logger.debug(
+        "barrier_wait[%s]: at barrier_tap for device %s. Thread %s",
+        logging_name, device, threading.current_thread()
+      )
     with lock:
       devices_at_barrier.append(device)
-      if logging.vlog_is_on(2):
+      if logger.isEnabledFor(logging.DEBUG):
```

Comment:
However note this one still is useful: it guards the computation of `waiting_for_devices`.